TYPO3.Flow/Classes/TYPO3/Flow/Mvc/ActionRequest.php

```diff
@@ -469,11 +469,12 @@ public function getControllerActionName()
      */
     public function setArgument($argumentName, $value)
     {
-        if (!is_string($argumentName) || strlen($argumentName) === 0) {
+        $argumentName = (string)$argumentName;
+        if (empty($argumentName)) {
             throw new Exception\InvalidArgumentNameException('Invalid argument name (must be a non-empty string).', 1210858767);
         }
 
-        if (substr($argumentName, 0, 2) === '__') {
+        if (strpos($argumentName, '__') === 0) {
```

Comment:
Was just a speed think that I apply whenever I pass by such a piece. We don't need the substr all we need to know is if the first two chars match. But I can change that back if you feel too uncomfortable with ti.